syntax = "proto2";

package dharma.common;

import "modules/canbus/proto/chassis.proto";
import "modules/localization/proto/pose.proto";

message VehicleState {
  optional double timestamp = 1 [default = 0.0];

  optional double x = 2 [default = 0.0];
  optional double y = 3 [default = 0.0];
  optional double z = 4 [default = 0.0];

  optional double heading = 5 [default = 0.0];
  optional double kappa = 6 [default = 0.0];

  optional double linear_velocity = 7 [default = 0.0];
  optional double angular_velocity = 8 [default = 0.0];
  optional double linear_acceleration = 9 [default = 0.0];
  optional dharma.localization.Pose pose = 10;
}
